kore/src/Kore/ASTVerifier/SentenceVerifier.hs

```diff
@@ -276,20 +277,37 @@ verifySymbolSentence indexedModule sentence =
             -- attribute record when it becomes available.
             isCtor =
                 Attribute.constructorAttribute  `elem` getAttributes attributes
-            resultSortHook = do
+            sortData = do
                 (sortDescription, _) <-
                     Map.lookup resultSortId
                         $ indexedModuleSortDescriptions indexedModule
-                Attribute.getHook . Attribute.Sort.hook $ sortDescription
-        in
-            koreFailWhen
-                (isCtor && isJust resultSortHook)
-                ( "Cannot define constructor '"
-                ++ getIdForError symbol
-                ++ "' for hooked sort '"
-                ++ getIdForError resultSortId
-                ++ "'."
-                )
+                let
+                    maybeHook =
```

Comment:
Since there are two different checks here, could we extract each one to its own top-level function, taking the `SortDescription` as input?